### ISSUE TITLE
resolv.conf no such file o directory 

### ISSUE THREAD

**Issue by danisilver** [moby/moby#4861]
i just left to install docker on ubuntu 12.04 from this guide : 
http://docs.docker.io/en/latest/installation/ubuntulinux/
and i got this

➜  ~  sudo docker run -i -t ubuntu /bin/bash
2014/03/26 11:45:17 Error: open /etc/resolv.conf: no such file or directory

any suggestions?

**Comment by duffsterlp** [comment]
@unclejack @crosbymichael I am still seeing this issue today on docker version 1.12.6.
Why does /etc/resolv.conf gate running docker containers?

**Comment by Robpol86** [comment]
I too am still seeing this on my fully patched (as of today) Fedora Server 25 system running "Docker version 17.03.1-ce, build c6d412e".

```
$ journalctl -b |grep -i resolv.conf
Apr 21 16:18:58 filesrv.rob86.net avahi-daemon[1328]: Failed to open /etc/resolv.conf: Invalid argument
Apr 21 16:19:01 filesrv.rob86.net dockerd[1451]: time="2017-04-21T16:19:01.397160570-07:00" level=error msg="Failed to start container b2111a98e8050c720d26c341358e5916543cee2cf29e6f21328d1925fa32ec45: open /etc/resolv.conf: no such file or directory"
Apr 21 16:19:01 filesrv.rob86.net dockerd[1451]: time="2017-04-21T16:19:01.445159230-07:00" level=error msg="Failed to start container 35ab16d33c1e6611b74f9a28b0872c4200cf0b122a2c4c1649b0737a9e13171a: open /etc/resolv.conf: no such file or directory"
Apr 21 16:19:01 filesrv.rob86.net dockerd[1451]: time="2017-04-21T16:19:01.485162302-07:00" level=error msg="Failed to start container e31ad6359d6a846ebce0e13aa2bdd706dafedb4efd8e757a9f26ceef659b8f07: open /etc/resolv.conf: no such file or directory"
Apr 21 16:19:01 filesrv.rob86.net dockerd[1451]: time="2017-04-21T16:19:01.521160844-07:00" level=error msg="Failed to start container cd56af1b4ff3a104a2584f4f89209b158030e6f883e30d5e5302efd3ee6b7a66: open /etc/resolv.conf: no such file or directory"
```

I mitigated this by running `sudo systemctl edit --full docker` and adding the following line to the `Service` section:

```
ExecStartPre=/bin/bash -c 'until [ -s /etc/resolv.conf ]; do sleep 1; done'
```

My fc25 server gets its IP and DNS from DHCP, nothing statically configured. My DHCP server is a pfSense box if that matters.

**Comment by runcom** [comment]
I think there hasn't been a good answer on this and since I'm also experiencing this I'm reopening. @thaJeztah could you ping network maintainers? At this point I don't understand why container start is gated by network (we do have isolated machines where there's no resolv.conf file but we want to start containers anyway)

**Comment by stefwalter** [comment]
The resolv.conf manual (from glibc) states:

       If this file does not exist, only the name server on the local
       machine will be queried; the domain name is determined from the
       hostname and the domain search path is constructed from the domain
       name.

Therefore it should be assumed that a Linux system without a resolv.conf is a valid Linux system.

Other software such as golang's DNS resolv.conf loading logic, deals with missing or inaccessible resolv.conf in the same way: https://golang.org/src/net/dnsconfig_unix.go

**Comment by martinpitt** [comment]
Also, resolv.conf is not necessary on systems which use non-libc NSS modules such as nss-resolve or winbind, ldap, etc.

**Comment by saimonn** [comment]
This bug also prevent from using docker offline on some distributions.

**Comment by hqhq** [comment]
Trying to fix it in: https://github.com/docker/libnetwork/pull/1847

**Comment by HeGuanyuan** [comment]
just create an empty file names 'resolv.conf' at the place, my problem solved~

**Comment by thaJeztah** [comment]
I see the fix for this was merged through https://github.com/moby/moby/commit/1ddeb11835600127a3319fc0dd3764e57ffbf521, which will be included in the upcoming 17.10 release; https://github.com/docker/docker-ce/blob/17.10/components/engine/vendor/github.com/docker/libnetwork/sandbox_dns_unix.go

closing, because this is resolved on master 👍
